Source/Core/UICommon/WiiUtils.cpp

```diff
@@ -27,12 +28,23 @@ bool InstallWAD(const std::string& wad_path)
   const auto es = ios.GetES();
 
   IOS::HLE::Device::ES::Context context;
-  if (es->ImportTicket(wad.GetTicket().GetBytes(), wad.GetCertificateChain()) < 0 ||
-      es->ImportTitleInit(context, tmd.GetBytes(), wad.GetCertificateChain()) < 0)
+  IOS::HLE::ReturnCode ret;
+  const bool checks_disabled = SConfig::GetInstance().m_disable_signature_checks;
+  while ((ret = es->ImportTicket(wad.GetTicket().GetBytes(), wad.GetCertificateChain())) < 0 ||
+         (ret = es->ImportTitleInit(context, tmd.GetBytes(), wad.GetCertificateChain())) < 0)
   {
+    if (!checks_disabled && ret == IOS::HLE::IOSC_FAIL_CHECKVALUE &&
+        AskYesNoT("This WAD does not have a valid signature. Continue to import?"))
```

Comment:
I think the first option is better, since it means the same thing and is completely correct, while being easier to understand. Thanks for the suggestion; I'll change this.